### ISSUE TITLE
network setup at install time

### ISSUE THREAD

**Issue by mateuszviste** [SvarDOS/bugz#20]
since SvarDOS relies heavily on networking (to access packages) it would make sense to try making the networking configuration as simple as possible to the user.

the major problem is drivers. install can ask the user what kind of NIC is installed in his PC, but then it also needs to find packet drivers somewhere... (that is - on the installation floppy). could maybe at least ship with the couple of the most common adapters (3COM, PC-NET, NE2000...).

then user could answer some simple form about networking: DHCP or static configuration, and in the latter case fill his IP, mask, gateway and DNS servers. All this would be used to preconfigure the wattcp.cfg file, install the appropriate pkt driver and put it into autoexec.bat.

**Comment by evan** [comment]
How about an optional 6th install floppy with all the drivers? User can say yes/no if they have it. If so, networking can be configured like you describe and the installer could also carry on to allow network install of optional packages.

**Comment by boeckmann** [comment]
A network setup utility would be VERY nice, at least for installing a packet driver. Perhaps as a separate menu-driven progam + floppy shipped with some drivers.

**Comment by mateuszviste** [comment]
> A network setup utility would be VERY nice, at least for installing a packet driver. Perhaps as a separate menu-driven progam + floppy shipped with some drivers.

I am not sure there is any advantage of this being a separate program. This could very well be performed by the SvarDOS installer, it is just an additional step where the user would have to choose a position from a "Network Interface Card" menu: NONE, 3COM 3C509-B, 3COM 3C905, NE-2000, etc.
If anything else than "NONE" is chosen, then at some later point the user would be asked to insert the "network drivers" floppy.

**Comment by boeckmann** [comment]
Yes, could of course be integrated in the installer. The only benefit in being a separate tool would be that it could be called after the system is installed, if for example the hardware changed. it could also be extended in the future to support other hardware, like mouse. Nothing that could not be integrated in the installer...

**Comment by bttrx** [comment]
Just rename the installer to setup.exe or svsetup.exe and make it dual-purpose, i.e., call it from `AUTOEXEC.BAT` during installation, but also let SvarDOS users run it manually after installation to change SvarDOS configuration.

But this approach is more complicated to develop, of course, because a user may already have fiddled around with `AUTOEXEC.BAT` and/or other relevant files.
